### PR TITLE
Upgrade to msbuild

### PR DESCRIPTION
**Pull request by jbagga**
Addresses #57 

cc @pranavkm

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -10,4 +10,6 @@ build_script:
   - build.cmd --quiet verify
 clone_depth: 1
 test: off
-deploy: off
\ No newline at end of file
+deploy: off
+# Required for dotnet-test to work
+os: Visual Studio 2015
```
Comment (pranavkm) [inline appveyor.yml]: 👏

File: JsonPatch.sln [hunk 2]
```diff
@@ -32,5 +45,5 @@ Global
 	GlobalSection(NestedProjects) = preSolution
 		{4D55F4D8-633B-462F-A5B1-FEB84BD2D534} = {430B59ED-F960-4D3A-8FFE-3370008E168D}
 		{81C20848-E063-4E12-AC40-0B55A532C16C} = {36CD6341-AB44-44EB-B3AA-BF98C89FECDD}
-	EndGlobalSection
+	EndGlobalSection\
```
Comment (pranavkm) [inline JsonPatch.sln]: Is the trailing slash intentional?

File: JsonPatch.sln [hunk 3]
```diff
@@ -1,15 +1,28 @@
...
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.AspNetCore.JsonPatch", "src\Microsoft.AspNetCore.JsonPatch\Microsoft.AspNetCore.JsonPatch.csproj", "{4D55F4D8-633B-462F-A5B1-FEB84BD2D534}"
 EndProject
-Project("{8BB2217D-0F2D-49D1-97BC-3654ED321F3B}") = "Microsoft.AspNetCore.JsonPatch.Test", "test\Microsoft.AspNetCore.JsonPatch.Test\Microsoft.AspNetCore.JsonPatch.Test.xproj", "{81C20848-E063-4E12-AC40-0B55A532C16C}"
+Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Solution Items", "Solution Items", "{C430C499-382D-47BD-B351-CF8F89C08CD2}"
+  	ProjectSection(SolutionItems) = preProject
+  		global.json = global.json
+ 		NuGet.config = NuGet.config
+  	EndProjectSection
+EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.AspNetCore.JsonPatch.Test", "test\Microsoft.AspNetCore.JsonPatch.Test\Microsoft.AspNetCore.JsonPatch.Test.csproj", "{81C20848-E063-4E12-AC40-0B55A532C16C}"
```
Comment (pranavkm) [inline JsonPatch.sln]: Could we move this line to 12 so there's fewer diff?

File: JsonPatch.sln [hunk 4]
```diff
@@ -1,15 +1,28 @@
...
 EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "test", "test", "{36CD6341-AB44-44EB-B3AA-BF98C89FECDD}"
 EndProject
-Project("{8BB2217D-0F2D-49D1-97BC-3654ED321F3B}") = "Microsoft.AspNetCore.JsonPatch", "src\Microsoft.AspNetCore.JsonPatch\Microsoft.AspNetCore.JsonPatch.xproj", "{4D55F4D8-633B-462F-A5B1-FEB84BD2D534}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.AspNetCore.JsonPatch", "src\Microsoft.AspNetCore.JsonPatch\Microsoft.AspNetCore.JsonPatch.csproj", "{4D55F4D8-633B-462F-A5B1-FEB84BD2D534}"
 EndProject
-Project("{8BB2217D-0F2D-49D1-97BC-3654ED321F3B}") = "Microsoft.AspNetCore.JsonPatch.Test", "test\Microsoft.AspNetCore.JsonPatch.Test\Microsoft.AspNetCore.JsonPatch.Test.xproj", "{81C20848-E063-4E12-AC40-0B55A532C16C}"
+Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Solution Items", "Solution Items", "{C430C499-382D-47BD-B351-CF8F89C08CD2}"
+  	ProjectSection(SolutionItems) = preProject
+  		global.json = global.json
```
Comment (kichalla) [inline JsonPatch.sln]: remove

**Pull request merged**